rust/routee-compass-powertrain/src/routee/prediction/interpolation/interpolation_speed_grade_model.rs

```diff
@@ -33,17 +33,26 @@ impl PredictionModel for InterpolationSpeedGradeModel {
         let grade_value = grade_unit.convert(&grade, &self.grade_unit).as_f64();
 
         // snap incoming speed and grade to the grid
-        let min_speed = self.interpolator.x[0].0;
-        let max_speed = self.interpolator.x[self.interpolator.x.len() - 1].0;
-        let min_grade = self.interpolator.y[0].0;
-        let max_grade = self.interpolator.y[self.interpolator.y.len() - 1].0;
+        let (min_speed, max_speed, min_grade, max_grade) = match &self.interpolator {
+            interp::Interpolator::Interp2D(interp) => (
+                interp.x[0],
+                *interp.x.last().unwrap(),
```

Comment:
Can we get rid of these unwraps and map the error into a `TraversalModelError`?

Reply:
You can add a new error here that is transparent and takes in whatever errors you're dealing with and it will automagically convert it into a `TraversalModelError` if you use the `?` operator.